control/controlclient/hostinfo_linux.go

```diff
@@ -99,5 +100,17 @@ func inContainer() (ret bool) {
 		}
 		return nil
 	})
+	lineread.File("/proc/mounts", func(line []byte) error {
+		if mem.Contains(mem.B(line), mem.S("fuse.lxcfs")) {
+			ret = true
+			return io.EOF
+		}
+		return nil
+	})
+	// https://cloud.google.com/run/docs/reference/container-contract#env-vars
+	if os.Getenv("K_REVISION") != "" && os.Getenv("K_CONFIGURATION") != "" &&
```

Comment:
I think this is notable enough to be worth calling out specially in Hostinfo somewhere. Like repeat lines 59-61 and add a new attribute, like `; env=kn` (env=knative). As this'll also fire on non-Cloud Run Knative environments.

Reply:
Done. On Cloud Run: `control: HostInfo: {"IPNVersion":"date.20210316","BackendLogID":"...","OS":"linux","OSVersion":"Alpine Linux v3.13; kernel=4.4.0; env=kn","Hostname":"google-assistant-mqtt","GoArch":"amd64","Services":[{"Proto":"tcp","Port":0,"Description":"tailscaled"}]}`

It isn't including an explicit "`; container`" but I think env=kn always implies it is running in a container.